db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -436,18 +435,20 @@ public boolean nextAccessPath(Optimizer optimizer,
                 boolean[] isAscending=irg.isAscending();
 
                 for(int i=0;i<baseColumnPositions.length;i++){
-                    int rowOrderDirection=isAscending[i]?RowOrdering.ASCENDING:RowOrdering.DESCENDING;
-                    int pos = rowOrdering.orderedPositionForColumn(rowOrderDirection,getTableNumber(),baseColumnPositions[i]);
-                    if (pos == -1) {
-                        rowOrdering.nextOrderPosition(rowOrderDirection);
-                        pos = rowOrdering.addOrderedColumn(rowOrderDirection,getTableNumber(),baseColumnPositions[i]);
-                    }
-                    // check if the column has a constant predicate like "col=constant" defined on it,
-                    // if so, we can treat it as sorted as it has only one value
-                    if (pos >=0 &&    /* a column ordering is added or exists */
-                        hasConstantPredicate(getTableNumber(), baseColumnPositions[i], predList)) {
-                        ColumnOrdering co = rowOrdering.getOrderedColumn(pos);
-                        co.setBoundByConstant(true);
+                    if (!irg.isOnExpression()) {
```

Comment:
Here we set proper ordering to base table columns according to index column ordering. For expression-based indexes, however, I'm not sure what to do here since whether an index column is ascending or descending doesn't really mean anything to the ordering of base table columns.

Reply:
OK, I created a Jira ticket to track this issue. I need to investigate a bit more on how we can respect the sort order defined on index expressions.